File: backend/src/main/java/com/team05/petmeeting/domain/campaign/dto/CampaignDetailRes.java
package com.team05.petmeeting.domain.campaign.dto;

import com.team05.petmeeting.domain.campaign.entity.Campaign;
import com.team05.petmeeting.domain.campaign.enums.CampaignStatus;

import java.util.List;

public record CampaignDetailRes (
        int campaignCount,
        List<CampaignDetailItem> campaigns
){
    public static CampaignDetailRes from(List<Campaign> campaign) {
        List<CampaignDetailItem> campaignDetailItems = campaign.stream()
                .map(CampaignDetailItem::from)
                .toList();
        return new CampaignDetailRes(campaignDetailItems.size(), campaignDetailItems);
    }
    public record CampaignDetailItem(
            Long id,
            String title,
            int targetAmount,
            int currentAmount,
            CampaignStatus status
    ){
        public static CampaignDetailItem from(Campaign campaign) {
            return new CampaignDetailItem(
                    campaign.getId(),
                    campaign.getTitle(),
                    campaign.getTargetAmount(),
                    campaign.getCurrentAmount(),
                    campaign.getStatus()
            );
        }
    }
}

File: backend/src/main/java/com/team05/petmeeting/domain/campaign/dto/CampaignDetailRes.kt
package com.team05.petmeeting.domain.campaign.dto

import com.team05.petmeeting.domain.campaign.entity.Campaign
import com.team05.petmeeting.domain.campaign.enums.CampaignStatus

@JvmRecord
data class CampaignDetailRes(
    val campaignCount: Int,
    val campaigns: MutableList<CampaignDetailItem?>?
) {
    @JvmRecord
    data class CampaignDetailItem(
        val id: Long?,
        val title: String?,
        val targetAmount: Int,
        val currentAmount: Int,
        val status: CampaignStatus?
    ) {
        companion object {
            fun from(campaign: Campaign): CampaignDetailItem {
                return CampaignDetailItem(
                    campaign.getId(),
                    campaign.getTitle(),
                    campaign.getTargetAmount(),
                    campaign.getCurrentAmount(),
                    campaign.getStatus()
                )
            }
        }
    }

    companion object {
        fun from(campaign: MutableList<Campaign?>): CampaignDetailRes {
            val campaignDetailItems = campaign.stream()
                .map<CampaignDetailItem?> { campaign: Campaign? -> CampaignDetailItem.Companion.from(campaign!!) }
                .toList()
            return CampaignDetailRes(campaignDetailItems.size, campaignDetailItems)
        }
    }
}
